Controls/Grid/Grid.UWP/View/Services/Editing/EditingService.cs

```diff
@@ -181,27 +181,29 @@ public bool CancelEdit(ActionTrigger trigger)
                     if (trigger != ActionTrigger.ExternalEditor)
                     {
                         this.Owner.ExternalEditor.CancelEdit();
+                        this.Owner.CurrencyService.RefreshCurrentItem(true);
                     }
                 }
                 else
                 {
                     this.Owner.Model.CancelEdit();
                     this.Owner.EditRowLayer.EditorLayoutSlots.Clear();
+                    this.Owner.CurrencyService.RefreshCurrentItem(true);
                 }
 
-                this.Owner.Model.CurrentDataProvider.CommitEditOperation(this.EditItem);
+                this.Owner.Model.CurrentDataProvider.CancelEditOperation(this.EditItem);
             })
             {
-                Flags = UpdateFlags.AffectsContent
+                Flags = UpdateFlags.AffectsData
             };
 
-            this.Owner.updateService.RegisterUpdate(update);
-
             foreach (var pair in this.operation.OriginalValues)
             {
                 pair.Key.SetValueForInstance(this.operation.EditItemInfo.Item, pair.Value);
             }
 
+            this.Owner.updateService.RegisterUpdate(update);
```

Comment:
Register the update after restoring the row's original values to ensure they are shown onscreen.